burnman/mineral.py

```diff
@@ -64,6 +65,15 @@ class which derives from the equation_of_state base class.
             raise Exception("Please derive your method from object (see python old style classes)")
 
         if self.method is not None and type(new_method) is not type(self.method):
+            # Reset everything but params
+            for key, elem in self.__dict__.items():
+                if key != 'params':
+                    elem = float('nan')
+
```

Comment:
Ok, let me try to get something together tomorrow that will be more explicit. I agree with your points, obviously what I've done here isn't the best solution (although set method itself wouldn't be a problem, as it is reset in the following line).